test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -434,8 +434,9 @@ void UcxContext::destroy_connections()
         _conn_requests.pop_front();
     }
 
-    for (conn_map_t::iterator iter = _conns.begin(); iter != _conns.end(); ++iter) {
-        delete iter->second;
+    UCX_LOG << "destroy_connections";
+    while (!_conns.empty()) {
+        delete _conns.begin()->second;
```

Comment:
seems like "double free", this way clean up needs pop_front/erase here and _conns.clear() below is redundant
this can re-balance the map on erase, so old code is better

Reply:
removed clear()
agree that it can cause redundant rebalance, but IMO it's simpler (than to move all connections to tmp list), and this code part is exit cleanup so perf (up to logn) should not be importatnt

Reply:
ok on rebalance impact but I still don't understand how does it work:
`delete _conns.begin()->second; ` deletes the `second` object but doesn't erase `begin()` element from the map, so `while (!_conns.empty()) {` is infinite loop which should cause multiple destruction/free of the same object

Reply:
the destructor of UcxConnection removes itself from the map
i'll add comment

Reply:
👍 so that means the old code was actually incorrect

Reply:
yes, it was incorrect